Source/WebCore/rendering/style/RenderStyle.cpp

```diff
@@ -1049,6 +1049,10 @@ static bool rareNonInheritedDataChangeRequiresLayerRepaint(const StyleRareNonInh
 #endif
 
     if (first.opacity != second.opacity) {
+        // If we're changing to/from 0 opacity, then repaint the layer since we try to skip buffer allocation
+        // for opacity:0.
```

Comment:
I think this needs a bit more explanation as paint invalidation and buffer allocation are highly unrelated at this level.

Reply:
Is there a better solution that lets us do this invalidation in a location that makes more sense?

I could add a new StyleDifferenceContextSensitiveProperty enum value, and then do the check in adjustStyleDifference?

Reply:
You need to do something that will result in `RenderLayer::setNeedsCompositingConfigurationUpdate()` getting called, probably via `RenderLayerCompositor::layerStyleChanged()`, so I'd expect a change between zero and non-zero opacity to result in a `StyleDifference::RecompositeLayer`.